ci: use automatic strategy selection for AngularTemplateCompile and TypeScriptCompile

josephperrott opened the pull request:
Removes the selection strategies for `AngularTemplateCompile` and `TypeScriptCompile` causing them to use the automatic selection strategies instead.


For `test_ivy_aot`
Before change:
`1919 processes: 1223 remote cache hit, 463 local, 233 remote`
After change:
`1919 processes: 1651 remote cache hit, 31 local, 237 remote`

josephperrott committed:
ci: use remote strategy for AngularTemplateCompile and TypeScriptCompile

devversion commented on .circleci/bazel.common.rc:
```diff
@@ -14,13 +14,7 @@ test --flaky_test_attempts=2
+# Utilize remote strategy for AngularTemplateCompile and TypeScriptCompile to take
+# advantage of caching of outputs
+build --strategy=AngularTemplateCompile=remote
+build --strategy=TypeScriptCompile=remote
```
Should we also rely on the Bazel's automatic spawn strategy resolution? Like we do in components?

josephperrott replied:
I think we can consider it in the future, but we have see problems with the worker strategy, and I am worried about it falling back to that strategy.  I feel like we need to come up with testing of it before we can count on it.

devversion replied:
Yeah I remember about these. I don't we ever hit those on `components` though? And sounds good, not really a priority for this PR 😄

josephperrott committed:
ci: add assets to ng_module in playground todo

josephperrott commented:
@alexeagle Did we ever sort out how to ensure that node is available on the path as noted here for our stamping?

https://github.com/angular/angular/blob/3a90a7cb43cefaceb234cbe29b56237a42e2fc14/.bazelrc#L44-L49

devversion reviewed (approved):
LGTM

devversion commented on .circleci/bazel.common.rc:
```diff
@@ -14,13 +14,7 @@ test --flaky_test_attempts=2
+# Utilize remote strategy for AngularTemplateCompile and TypeScriptCompile to take
+# advantage of caching of outputs
+build --strategy=AngularTemplateCompile=remote
+build --strategy=TypeScriptCompile=remote
```
Actually, do we need to have a fall-back strategy? I see some CI failures and I guess it's because the `npm_package` rules are marked as local-only.

josephperrott replied:
Updated to use the automatic strategy selection.  Can you give it another look now?

devversion replied:
Looks good to me! :100: Nit: that the commit message could say that it sets up the automatic selection as nothing got _added_.

josephperrott committed:
ci: use automatic selection strategies

Removes the selection strategies for `AngularTemplateCompile` and `TypeScriptCompile` causing
them to use the automatic selection strategies instead.

josephperrott commented:
Merge-assistance: global approval

Talked to Igor offline, confirmed these changes under  `modules/` as they currently are ownerless

AndrewKushnir commented:
@josephperrott there was a conflict while merging to patch branch, so I merged to master only for now. Could you plz create a separate PR targeting patch branch? Thank you.
